### ISSUE TITLE
MSL : hlsl that has nested structs with arrays fails to convert to msl

### ISSUE THREAD

**Issue by fatkas** [KhronosGroup/SPIRV-Cross#457]
this source 
https://gist.github.com/fatkas/7eb5cdfb5b96ebd7549f1bf005f35536
fails to convert to msl with "Struct member does not have ArrayStride set." error. Looks like that's because of nested struct but i'm not sure how do i fix it, any help would be appreciated. Thanks!

**Comment by JJoosten** [comment]
We see the same issue when coming from GLSL; compilation started to go wrong for us after the following commit: https://github.com/KhronosGroup/SPIRV-Cross/commit/607b0d6d42583a199b28bfd7f3200ed3fa6a591d 
Cheers!

**Comment by billhollings** [comment]
@JJoosten 

Can you submit a GLSL shader that illustrates the problem please?

**Comment by JJoosten** [comment]
@billhollings sure! :)

Just quickly typed up and tested this shader (it illustrates the problem) (for us its actually: "Struct member does not have Offset set.").

Cheers,

Callstack:
- spirv_cross::Compiler::type_struct_member_offset(spirv_cross::SPIRType const&, unsigned int) const at spirv_cross.cpp:2238
- spirv_cross::Compiler::get_declared_struct_size(spirv_cross::SPIRType const&) const spirv_cross.cpp:2266
- spirv_cross::CompilerMSL::get_declared_struct_member_size(spirv_cross::SPIRType const&, unsigned int) const at spirv_msl.cpp:3645
- spirv_cross::CompilerMSL::align_struct(spirv_cross::SPIRType&) at spirv_msl.cpp:869
- spirv_cross::CompilerMSL::emit_resources() at spirv_msl.cpp:1334
- spirv_cross::CompilerMSL::compile() at spirv_msl.cpp:181

```	
#version 450
				   
struct s1
{
	int a;
};

struct s2
{
	s1 b;
};

layout(std430, binding = 1) buffer dstbuffer{ s2 test[]; };
layout(local_size_x = 1, local_size_y = 1, local_size_z = 1) in;
void main()
{
	s2 testVal;
	testVal.b.a = 0;
	test[0] = testVal;
}
```

**Comment by billhollings** [comment]
Fixed in PR #464.
